app/forms/portal/client_login_form.rb

```diff
@@ -14,22 +14,20 @@ def client
     private
 
     def matches_client
-      if last_four.present?
+      if number.present?
         @client = possible_clients.find do |client|
-          primary_match = ActiveSupport::SecurityUtils.secure_compare(client.intake&.primary_last_four_ssn.to_s, last_four)
-          spouse_match = ActiveSupport::SecurityUtils.secure_compare(client.intake&.spouse_last_four_ssn.to_s, last_four)
-          primary_match || spouse_match
+          primary_match = ActiveSupport::SecurityUtils.secure_compare(client.intake&.primary_last_four_ssn.to_s, number)
```

Comment:
instead of checking all three, could we return as soon as we find a match?

Reply:
Yes - we could return as soon as we find a match. Good call.

Originally I was thinking that it'd be nice to always check all 3 is to avoid leaking any information about _which_ one matched, if you get it right. I've been taught to be scared of [timing attacks on the web.](http://crypto.stanford.edu/~dabo/papers/webtiming.pdf)

In this case, I guess you've submitted the correct value, so now you're logged in, so in the long run, we're going to permit you to view the client's data, and therefore you're authorized to click around in the app and find out which one you matched, so it's OK to exit early.

Reply:
Pushed in https://github.com/codeforamerica/vita-min/commit/ca4aa2077462d202b36f4879f10f0509d93ed9e5